agent/stats/engine.go

```diff
@@ -395,6 +418,23 @@ func (engine *DockerStatsEngine) getContainerMetricsForTask(taskArn string) ([]*
 	return containerMetrics, nil
 }
 
+func (engine *DockerStatsEngine) doRemoveContainer(container *StatsContainer, taskArn string) {
+	container.StopStatsCollection()
+	dockerID := container.containerMetadata.DockerID
+	delete(engine.tasksToContainers[taskArn], dockerID)
+	seelog.Debugf("Deleted container from tasks, id: %s", dockerID)
+
+	if len(engine.tasksToContainers[taskArn]) == 0 {
```

Comment:
There's no need to remove the entry for the task if its composed of more than 1 container, this would be handled when that condition becomes true.